gitrevise/merge.py

```diff
@@ -199,29 +202,131 @@ def merge_blobs(
         if err.returncode < 0:
             raise
 
+        preimage = err.output
+        recorded_postimage, conflict_id = reuse_recorded_resolution(repo, preimage)
+        if recorded_postimage:
+            return recorded_postimage
+
         # At this point, we know that there are merge conflicts to resolve.
         # Prompt to try and trigger manual resolution.
         print(f"Conflict applying '{labels[2]}'")
         print(f"  Path: '{path}'")
         if input("  Edit conflicted file? (Y/n) ").lower() == "n":
-            raise MergeConflict("user aborted")
+            raise MergeConflict("user aborted")  # pylint: disable=W0707
 
         # Open the editor on the conflicted file. We ensure the relative path
         # matches the path of the original file for a better editor experience.
         conflicts = tmpdir / "conflict" / path.relative_to("/")
         conflicts.parent.mkdir(parents=True, exist_ok=True)
-        conflicts.write_bytes(err.output)
+        conflicts.write_bytes(preimage)
         merged = edit_file(repo, conflicts)
 
         # Print warnings if the merge looks like it may have failed.
-        if merged == err.output:
+        if merged == preimage:
             print("(note) conflicted file is unchanged")
 
-        if b"<<<<<<<" in merged or b"=======" in merged or b">>>>>>>" in merged:
+        if (
+            b"<" * MARKER_SIZE in merged
+            or b"=" * MARKER_SIZE in merged
+            or b">" * MARKER_SIZE in merged
+        ):
             print("(note) conflict markers found in the merged file")
 
         # Was the merge successful?
         if input("  Merge successful? (y/N) ").lower() != "y":
-            raise MergeConflict("user aborted")
+            raise MergeConflict("user aborted")  # pylint: disable=W0707
+
+        record_resolution(repo, conflict_id, preimage, merged)
 
     return Blob(current.repo, merged)
+
+
+def reuse_recorded_resolution(repo, preimage) -> Tuple[Optional[Blob], Optional[str]]:
+    rr_cache = repo.gitdir / "rr-cache"
+    # TODO Should we cache settings like this?
+    if not repo.bool_config("rerere.enabled", default=rr_cache.is_dir()):
+        return None, None
```

Comment:
It might be good to also have a `revise.rerere` config value which we also check, and defaults to the standard rerere config value, so that if someone runs into an issue with git-revise's rerere implementation, they can turn that off without turning off rerere in general.

We can also consider caching this on the `Repository` in the future, but I don't think that's necessary in this initial PR.